mkosi/distributions/serpentos.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import re
+import shutil
+import textwrap
+from collections.abc import Sequence
+
+from mkosi.config import Architecture
+from mkosi.context import Context
+from mkosi.distributions import Distribution, DistributionInstaller, PackageType
+from mkosi.installer.moss import invoke_moss
+from mkosi.log import die
+
+
+class Installer(DistributionInstaller):
+    @classmethod
+    def pretty_name(cls) -> str:
+        return "Serpent OS"
+
+    @classmethod
+    def filesystem(cls) -> str:
+        return "btrfs"
+
+    @classmethod
+    def package_type(cls) -> PackageType:
+        return PackageType.moss
+
+    @classmethod
+    def default_release(cls) -> str:
+        return "rolling"
+
+    @classmethod
+    def default_tools_tree_distribution(cls) -> Distribution:
+        return Distribution.serpentos
+
+    @classmethod
+    def _move_kernel_and_initrd(cls, context: Context) -> None:
+        regexp = re.compile("com\.serpentos\.([a-z][a-z0-9]+)\.(.*)")
+
+        # SerpentOS puts its kernels and initrds into /usr/lib/kernel
+        kernel_dir = context.root / "usr/lib/kernel"
+        for k in kernel_dir.glob("com.serpentos.*"):
+            match = regexp.fullmatch(str(k.name))
+            if not match:
+                continue
+
+            kkind = match.group(1)
+            kver = match.group(2)
+
+            modules_dir = (context.root / f"usr/lib/modules/{kver}.{kkind}")
+            if modules_dir.is_dir():
+                shutil.move(k, (modules_dir / f"vmlinuz-{kver}.{kkind}"))
+                shutil.move((kernel_dir / f"initrd-com.serpentos.{kkind}.{kver}"),
+                            (modules_dir / f"initrd-{kver}.{kkind}"))
```

Comment:
No need to move the initrd since we won't do anything with it.

Reply:
To clarify, having to move the files at all is a hack. Having kernel related files in /usr/lib/modules is practically Linux API for all intents and purposes and SerpentOS should really be installing its kernels and such in there by default. Doing so will save them a lot of pain getting all kinds of tools working that will look for this stuff in /usr/lib/modules.

But sure, feel free to move all files if you care about this.